src/InitializeData/FetchData.ts

```diff
@@ -29,13 +32,60 @@ export async function fetchLocalStoreageData() {
 }
 
 export async function fetchCrawlData() {
-  const [cafeRes, martRes, mealListRes, mealDormListRes, mealMenuListRes] =
-    await Promise.all([
-      axios.get('https://snuco.snu.ac.kr/ko/node/21'),
-      axios.get('https://snuco.snu.ac.kr/ko/node/19'),
-      axios.get('https://snuco.snu.ac.kr/ko/node/20'),
-      axios.get('https://snudorm.snu.ac.kr/food-schedule/'),
-      axios.get('https://snuco.snu.ac.kr/ko/foodmenu'),
-    ]);
-  return {cafeRes, martRes, mealListRes, mealDormListRes, mealMenuListRes};
+  // const now = new Date('Tue Oct 26 2021 12:24:15 GMT+0900');
+  const now = new Date();
+  const year = getYear(now);
+  const month = getMonth(now) + 1;
+  const date = getDate(now);
+  const day0MenuURL = `https://snuco.snu.ac.kr/ko/foodmenu?field_menu_date_value_1%5Bvalue%5D%5Bdate%5D=&field_menu_date_value%5Bvalue%5D%5Bdate%5D=${month}%2F${date}%2F${year}`;
+
+  function getDateMenuURL(
+    thisYear: number,
+    thisMonth: number,
+    thisDate: number,
+    offset: number,
+  ): string {
+    const _day = new Date(thisYear, thisMonth - 1, thisDate + offset);
+    const _year = getYear(_day);
+    const _month = getMonth(_day) + 1;
+    const _date = getDate(_day);
+    const _url = `https://snuco.snu.ac.kr/ko/foodmenu?field_menu_date_value_1%5Bvalue%5D%5Bdate%5D=&field_menu_date_value%5Bvalue%5D%5Bdate%5D=${_month}%2F${_date}%2F${_year}`;
+    return _url;
+  }
+  const day1MenuURL = getDateMenuURL(year, month, date, 1);
+  const day2MenuURL = getDateMenuURL(year, month, date, 2);
+  const day_1MenuURL = getDateMenuURL(year, month, date, -1);
+  const day_2MenuURL = getDateMenuURL(year, month, date, -2);
+  const [
+    cafeRes,
+    martRes,
+    mealListRes,
+    mealDormListRes,
+    day0MenuListRes,
+    day1MenuListRes,
+    day2MenuListRes,
+    day_1MenuListRes,
+    day_2MenuListRes,
+  ] = await Promise.all([
+    axios.get('https://snuco.snu.ac.kr/ko/node/21'),
+    axios.get('https://snuco.snu.ac.kr/ko/node/19'),
+    axios.get('https://snuco.snu.ac.kr/ko/node/20'),
+    axios.get('https://snudorm.snu.ac.kr/food-schedule/'),
+    axios.get(day0MenuURL),
+    axios.get(day1MenuURL),
+    axios.get(day2MenuURL),
+    axios.get(day_1MenuURL),
+    axios.get(day_2MenuURL),
```

Comment:
map을 이용하여 데이터 정제 흐름의 시작부터 끝까지 array 타입을 유지하면 원하는 날짜 범위에 대해 일반화된 구조를 만들 수 있습니다.
```suggestion
    axios.get('https://snuco.snu.ac.kr/ko/node/21'),
    axios.get('https://snuco.snu.ac.kr/ko/node/19'),
    axios.get('https://snuco.snu.ac.kr/ko/node/20'),
    axios.get('https://snudorm.snu.ac.kr/food-schedule/'),
    Promise.all(menuUrlsWithOffset.map(async ({offset,url}) => {
    const res = await axios.get(url);
    return { res,offset};
    })))
```

Reply:
이렇게 바꾸고 fetchCrawlData() 자체의 리턴에 offset을 포함해서 주라는 것?

Reply:
offset 또는 년,월,일
년, 월, 일 을 포함하면 앱을 보다가 날짜가 바뀌어버리는 경우를 대응할 수 있게 됨